File: examples/thymeleaf-java-example/src/main/java/de/tschuehly/spring/viewcomponent/thymeleaf/application/web/simple/SimpleViewComponent.java
package de.tschuehly.spring.viewcomponent.thymeleaf.application.web.simple;

import de.tschuehly.spring.viewcomponent.core.IViewContext;
import de.tschuehly.spring.viewcomponent.core.component.ViewComponent;
import de.tschuehly.spring.viewcomponent.thymeleaf.application.core.ExampleService;

@ViewComponent
public class SimpleViewComponent {
    final ExampleService exampleService;

    public SimpleViewComponent(ExampleService exampleService) {
        this.exampleService = exampleService;
    }

    record SimpleView(String helloWorld)implements IViewContext{}

    public SimpleView render() {
        return new SimpleView(exampleService.getHelloWorld());
    }
}

File: examples/thymeleaf-kotlin-example/src/main/kotlin/de/tschuehly/spring/viewcomponent/thymeleaf/application/web/simple/SimpleViewComponent.kt
package de.tschuehly.spring.viewcomponent.thymeleaf.application.web.simple

import de.tschuehly.spring.viewcomponent.core.IViewContext
import de.tschuehly.spring.viewcomponent.core.action.PostViewAction
import de.tschuehly.spring.viewcomponent.core.component.ViewComponent
import de.tschuehly.spring.viewcomponent.thymeleaf.application.core.ExampleService

@ViewComponent
class SimpleViewComponent(
    val exampleService: ExampleService
) {
    fun render() = HelloWorldView(exampleService.getHelloWorld())

    data class HelloWorldView(val helloWorld: String) : IViewContext

    @PostViewAction
    fun testAction(): IViewContext {
        return render()
    }
}